javascript/net/grpc/web/generator/build.zig

```diff
@@ -0,0 +1,215 @@
+const std = @import("std");
+const CrossTarget = std.zig.CrossTarget;
+
+fn format(comptime fmt: []const u8, args: anytype) []const u8 {
+    return std.fmt.allocPrint(std.testing.allocator, fmt, args) catch unreachable;
+}
+
+const BinaryTarget = struct {
+    name: []const u8,
+    arch: []const u8,
+};
+
+pub fn build(b: *std.build.Builder) void {
+    // Standard release options allow the person running `zig build` to select
+    // between Debug, ReleaseSafe, ReleaseFast, and ReleaseSmall.
+    const mode = b.standardReleaseOptions();
+
+    var version = if (std.os.getenv("VERSION")) |v| v else "unknown";
+
+    var targets = [_]BinaryTarget{
+        // for now, let's only build aarch64 binaries
+        // .{ .name = format("protoc-gen-grpc-web-{s}-linux-x86_64", .{version}), .arch = "x86_64-linux" },
+        // .{ .name = format("protoc-gen-grpc-web-{s}-darwin-x86_64", .{version}), .arch = "x86_64-macos" },
+        // .{ .name = format("protoc-gen-grpc-web-{s}-windows-x86_64", .{version}), .arch = "x86_64-windows" },
+
+        .{ .name = format("protoc-gen-grpc-web-{s}-linux-aarch64", .{version}), .arch = "aarch64-linux" },
+        .{ .name = format("protoc-gen-grpc-web-{s}-darwin-aarch64", .{version}), .arch = "aarch64-macos" },
+        .{ .name = format("protoc-gen-grpc-web-{s}-windows-aarch64", .{version}), .arch = "aarch64-windows" },
+    };
+
+    for (targets) |target| {
+        const exe = b.addExecutable(target.name, "grpc_generator.cc");
+        exe.linkLibCpp();
+        exe.linkSystemLibrary("pthread");
+        exe.linkSystemLibrary("dl");
+        exe.addIncludeDir("../../../../../third_party/protobuf/src");
+        exe.defineCMacro("HAVE_PTHREAD", "1");
+        exe.addCSourceFiles(&[_][]const u8{
+            // libprotobuf_lite
+            "../../../../../third_party/protobuf/src/google/protobuf/any_lite.cc",
```

Comment:
I copied the list from `third_party/protobuf/cmake/libprotobuf-lite.cmake`, `third_party/protobuf/cmake/libprotobuf.cmake` and `third_party/protobuf/cmake/libprotoc.cmake`, without any modifications.

I agree it is a pain that we need to keep the list up-to-date, but it is necessary when doing cross-compiling(we need to compile dependencies from source code as well).

Reply:
By any chance this can be parsed dynamically from the cmake files? If not, could you add a comment on the top of the list on how it should be maintained (and potentially improved)? Thanks!

Reply:
It is possible, but personally, I think the complexity is too high for a build script (Do we need to write tests for it?), so I would prefer to copy the list manually.